blocks/components/editable/index.js

```diff
@@ -101,14 +110,26 @@ export default class Editable extends wp.element.Component {
 		this.props.onFocus();
 	}
 
-	onChange() {
-		if ( ! this.editor.isDirty() ) {
+	isActive() {
+		return tinymce.EditorManager.activeEditor.id === this.editor.id;
+	}
+
+	onSelectionChange() {
+		// We must check this because selectionChange is a global event.
+		if ( ! this.isActive() ) {
 			return;
 		}
 
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.savedContent );
+		this.content = this.getContent();
+		this.selection = this.getSelection();
+
+		if ( ! this.props.onChange ) {
+			return;
+		}
+
+		if ( ! isEqual( this.props.value, this.content ) ) {
```

Comment:
It is not always equal, actually it will be more unequal as equal. No need to call onChange needlessly. Re `selectionChange`, I've found it be be one of the best events to pick up content changes. It always work regardless of the input method.